integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/merge/CarbonMergeDataSetCommand.scala

```diff
@@ -194,29 +210,32 @@ case class CarbonMergeDataSetCommand(
             tuple._2.asJava)
         }
       }
-      Some(UpdateTableModel(true, trxMgr.getLatestTrx,
-        executorErrors, tuple._2, true))
+      Some(UpdateTableModel(isUpdate = true, trxMgr.getLatestTrx,
+        executorErrors, tuple._2, loadAsNewSegment = true))
     } else {
       None
     }
 
-    CarbonInsertIntoWithDf(
-      databaseNameOp = Some(carbonTable.getDatabaseName),
+    val dataFrame = loadDF.select(tableCols.map(col): _*)
+    CarbonInsertIntoCommand(databaseNameOp = Some(carbonTable.getDatabaseName),
       tableName = carbonTable.getTableName,
-      options = Map("fileheader" -> header, "sort_scope" -> "nosort"),
+      options = Map("fileheader" -> header, "sort_scope" -> "no_sort"),
```

Comment:
I know it is base behvior, is it better to use target table sort scope ?

Reply:
if we focus on the cdc performance, existing no sort is fine. As your point, we cant blindly go with target table's sort scope, because it can happen that the target table has the sort scope has no sort.

Reply:
I agree with @ajantha-bhat  we should use the target table sort scope. But this PR is only about performance we can cover this change in another PR with proper test

Reply:
@akashrn5 : instead of fixing no_sort, would have analyzed why it was added. Originally it was using Target table sort scope only. #3764 added this. would have removed it here.
Now that you changed to no_sort, target table sort_scope is not used.

Reply:
check this.
https://github.com/apache/carbondata/pull/3901